### PR TITLE
Carousel: Add styling to Arrow Buttons, Dot nav, and make visibleSlides responsive

### PR DESCRIPTION
**Pull request by ryan-hunt-priceline**
New Props:
- sliceSpacing: Padding between slides (use DS ints)
- arrowPosition: top, side, bottom, hide - location of the nav arrows

Other updates:
- visibleSlides is responsive with 3 breakpoints
- mobile browsers will hide arrows and use css scroll
- Arrow button style updates
- navigation dot style updates

### CODE REVIEW

File: packages/carousel/src/Carousel.js [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-
+import { CarouselProvider, Slide } from 'pure-react-carousel'
 import { layoutToFlexWidths } from './layoutToFlexWidths'
 import { CarouselWrapper } from './Carousel.styles'
-
-const getSlideKey = moize(uuidv4)
+import { Flex, Relative } from 'pcln-design-system'
+import { Dots } from './Dots/Dots'
+import { ArrowButton } from './ArrowButton/ArrowButton'
+import { Slider } from './Slider/Slider'
```
Comment (pcln-james) [inline packages/carousel/src/Carousel.js:9]: if you export these from an `index.js` file then you don't need the duplication in the import path

File: packages/carousel/src/ArrowButton/ArrowButton.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          <ButtonNext>
+            <ChevronRight size={type === 'side' ? 40 : 32} mr='-3px' />
+          </ButtonNext>
+        )}
+      </Wrapper>
+    </Hide>
+  ) : null
+
+ArrowButton.propTypes = {
+  type: PropTypes.oneOf(['back', ' next']),
```
Comment (craigpalermo) [inline packages/carousel/src/ArrowButton/ArrowButton.js:26]:
```suggestion
  type: PropTypes.oneOf(['back', 'next']),
```

Comment (craigpalermo) [reply packages/carousel/src/ArrowButton/ArrowButton.js:26]: What do you think about `next` and `previous`?

File: packages/carousel/src/ArrowButton/ArrowButton.styles.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import styled from 'styled-components'
+import { getPaletteColor, Box } from 'pcln-design-system'
+
+const defaultStyles = (props) => `
+  padding: 0;
+  background: ${getPaletteColor('background.lightest')(props)};
+  border-radius: 100%;
+  border: none;
+  box-shadow: ${props.theme.shadows.md};
```
Comment (craigpalermo) [inline packages/carousel/src/ArrowButton/ArrowButton.styles.js:9]: @wesleymartin85 should this use the new box-shadow theme values?

File: packages/carousel/src/ArrowButton/ArrowButton.styles.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import styled from 'styled-components'
+import { getPaletteColor, Box } from 'pcln-design-system'
+
+const defaultStyles = (props) => `
+  padding: 0;
+  background: ${getPaletteColor('background.lightest')(props)};
+  border-radius: 100%;
```
Comment (craigpalermo) [inline packages/carousel/src/ArrowButton/ArrowButton.styles.js:7]: This seems like a good use case for `rounded='full'`!

File: packages/carousel/src/Carousel.js [hunk 5]
```diff
@@ -22,17 +21,35 @@ export const Carousel = ({
...
   const widths = layoutToFlexWidths(layout, children.length)
   const layoutSize = layout?.split('-').length
 
+  const useResponsiveVisibleSlides = (visibleSlides) => {
+    const [windowWidth, setWindowWidth] = useState(window.innerWidth)
+    useEffect(() => {
+      const handleResize = () => {
+        setWindowWidth(window.innerWidth)
+      }
+      window.addEventListener('resize', handleResize)
```
Comment (craigpalermo) [inline packages/carousel/src/Carousel.js:36]: Should this be debounced at all?

Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js:36]: Might be a good idea - were you thinking `lodash` debounce?

File: packages/carousel/src/Carousel.js (hunk 1 above)
Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js:9]: I hate index files haha, but I can add those in

Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js]: Updated :)

File: packages/carousel/src/Carousel.js (hunk 5 above)
Comment (pcln-james) [reply packages/carousel/src/Carousel.js:36]: we could separate this hook out to another file to make it easier to read/test

Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js:36]: I originally had this in a separate file, but I was getting a code scan error saying it must be defined inside a component

File: packages/carousel/src/ArrowButton/ArrowButton.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { ButtonBack, ButtonNext } from 'pure-react-carousel'
+import { ChevronRight, ChevronLeft } from 'pcln-icons'
+import { Hide } from 'pcln-design-system'
+import { Wrapper } from './ArrowButton.styles'
+
+const ArrowButton = ({ type, position, setPosition, ...props }) =>
+  setPosition === position ? (
+    <Hide xs sm>
+      <Wrapper type={type} position={position} {...props}>
+        {type === 'back' ? (
```
[the remark this answers is not in the document]
Comment (ryan-hunt-priceline) [reply packages/carousel/src/ArrowButton/ArrowButton.js:12]: Yeah, I think `previous` is better verbiage too

[the remark this answers is not in the document]
Comment (ryan-hunt-priceline) [reply packages/carousel/src/ArrowButton/ArrowButton.js:12]: This is not an external prop though (just used at the subcomponent level), so we can update this whenever

[the remark this answers is not in the document]
Comment (pcln-james) [reply packages/carousel/src/ArrowButton/ArrowButton.js:12]: sounds good. I'll hold you to updating it in a future PR 😛

File: packages/carousel/src/Carousel.js (hunk 5 above)
Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js]: I was able to pull this file out and pass the scan, it just has to have `use` to begin the function name

Comment (pcln-james) [reply packages/carousel/src/Carousel.js]: nice! yeah that makes sense

**Comment by pcln-james** [comment]
Could you add a description to this PR for future reference please. Don't link your team's Jira ticket because this is a public repo and won't help much lol

File: packages/carousel/src/ArrowButton/ArrowButton.styles.js (hunk 3 above)
Comment (ryan-hunt-priceline) [reply packages/carousel/src/ArrowButton/ArrowButton.styles.js]: Updated

**Review by craigpalermo — APPROVED** [review]

File: packages/carousel/src/Carousel.stories.js [hunk 7]
```diff
@@ -44,13 +48,30 @@ export default {
...
-          Slide {idx}
+      <ToutCard height='300px' borderRadius={20} boxShadowSize='md' borderWidth={0}>
+        <BackgroundImage height='190px' width='100%' image='https://cdn2.thecatapi.com/images/dnn.jpg' />
+        <Flex color='background.lightest' p={[1, 1, 2, 2, 2, 3]}>
+          <Box>
+            <Text fontSize={[0, 0, 1, 1, 2]} fontWeight='bold' mb={2}>
+              City {idx}, AB
+            </Text>
+            <Text fontSize={[0, 0, 0, 0, 1]}>Hotel for 2 nights</Text>
+            <Text fontSize={[0, 0, 0, 0, 1]}>+ Round-trip flight</Text>
```
Comment (pcln-james) [inline packages/carousel/src/Carousel.stories.js:59]: should we use `[0, null, null, null, 1]`? It's more long winded but it creates less CSS media queries which could impact performance

Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.stories.js:59]: Updated

**Comment by wesleymartin85** [comment]
Screen Shot 2022-01-31 at 3 40 40 PM
my only feedback is on the overall component's container, and how the slides will align with other content on the page. In the example above you'll see how the slides will align with a layout of three cards. It's off by 8px when arrows are hidden, top and bottom (not that big a deal), but this issue gets more visible when arrows are on the side.

**Review by pcln-james — APPROVED** [review]

**Comment by ryan-hunt-priceline** [comment]
> Screen Shot 2022-01-31 at 3 40 40 PM my only feedback is on the overall component's container, and how the slides will align with other content on the page. In the example above you'll see how the slides will align with a layout of three cards. It's off by 8px when arrows are hidden, top and bottom (not that big a deal), but this issue gets more visible when arrows are on the side.

Just pushed updates to address this

**Pull request merged**